trieste/acquisition/rule.py

```diff
@@ -1097,6 +1104,225 @@ def state_func(
         return state_func
 
 
+class TURBO(
+    AcquisitionRule[types.State[Optional["TURBO.State"], TensorType], Box, SupportsGetKernel]
+):
+    """Implements the TURBO algorithm as detailed in :cite:`eriksson2019scalable,`."""
+
+    @dataclass(frozen=True)
+    class State:
+        """The acquisition state for the :class:`TURBO` acquisition rule."""
+
+        acquisition_space: Box
+        """ The search space. """
+
+        L: float
+        """ Length of the trust region (before standardizing by model lengthscales) """
+
+        failure_counter: int
+        """ Number of consecutive failures (reset if we see a success). """
+
+        success_counter: int
+        """ Number of consecutive successes (reset if we see a failure).  """
+
+        y_min: TensorType
+        """ The minimum observed value. """
+
+        def __deepcopy__(self, memo: dict[int, object]) -> TURBO.State:
```

Comment:
I suspect this isn't necessary (though I'm aware you just copied it from the other trust region implementation). As far as I can see box_copy isn't actually a copy (due to box's override on `__deepcopy__`) and the returned `TURBO.State` is just what the default deepcopy would have done anyway?

If you're feeling virtuous you could remove the deepcopy and see whether the tests still all pass. Otherwise feel free to leave as is.